### PR TITLE
Remove unqueryable lists from admin meta

### PR DESCRIPTION
**Pull request by gwyneplaine**
update createAdminMeta to not include lists with graphql querying disabled. also apply this to relationships to those omitted lists

https://user-images.githubusercontent.com/12119389/132619815-a33f6139-df7d-4d0d-8897-0230969b2c61.mov

**PLEASE FINE TOOTH COMB THIS PR**
As I've operated here with half context at best.

### CODE REVIEW

File: packages/keystone/src/admin-ui/system/createAdminMeta.ts [hunk 1]
```diff
@@ -98,6 +108,8 @@ export function createAdminMeta(
     }
 
     for (const [fieldKey, field] of Object.entries(list.fields)) {
+      // If the field is a relationship field and is related to an omitted list, skip.
+      if (field.dbField.kind === 'relation' && omittedLists.includes(field.dbField.list)) continue;
```
Comment (gwyneplaine) [inline packages/keystone/src/admin-ui/system/createAdminMeta.ts:112]: We want to check if the field is a relationship to an omitted list, is interrogating the dbField on said list the right way to make this check? @timleslie

File: packages/keystone/src/admin-ui/system/createAdminMeta.ts [hunk 2]
```diff
@@ -16,8 +16,17 @@ export function createAdminMeta(
     views: [],
   };
 
+  const omittedLists: string[] = [];
+
   for (const [key, list] of Object.entries(initialisedLists)) {
     const listConfig = lists[key];
+    if (list.graphql.isEnabled.query === false) {
```
Comment (gwyneplaine) [inline packages/keystone/src/admin-ui/system/createAdminMeta.ts:23]: is `isEnabled.query` the only flag we want to skip on here?

**Review by mitchellhamilton — DISMISSED** [review]
I need to look at things

File: packages/keystone/src/admin-ui/system/createAdminMeta.ts (hunk 2 above)
Comment (mitchellhamilton) [reply packages/keystone/src/admin-ui/system/createAdminMeta.ts:23]: As a separate thing, if `list.graphql.isEnabled.create`/`list.graphql.isEnabled.delete` is false, we should force hideCreate/hideDelete to return false https://github.com/keystonejs/keystone/blob/3cfc2a3839142dd3ccdbf1dd86768257e9acc0dc/packages/keystone/src/admin-ui/system/getAdminMetaSchema.ts#L204-L227. We should also force the itemView fieldMode to read for all the fields of a list when the list has update omitted so you can't get a confusing error by editing and then trying to save (doing that is in the same file as above)

While we could technically make the creating in the Admin UI work for a list that has querying disabled so arguably the above check should be `isEnabled.type`, we definitely shouldn't do that now so `isEnabled.query` is correct imo.

Comment (timleslie) [reply packages/keystone/src/admin-ui/system/createAdminMeta.ts:23]: Yes.

File: packages/keystone/src/admin-ui/system/createAdminMeta.ts (hunk 1 above)
Comment (timleslie) [reply packages/keystone/src/admin-ui/system/createAdminMeta.ts:112]: Yep

File: packages/keystone/src/admin-ui/system/createAdminMeta.ts (hunk 2 above)
Comment (gwyneplaine) [reply packages/keystone/src/admin-ui/system/createAdminMeta.ts:23]: > As a separate thing, if `list.graphql.isEnabled.create`/`list.graphql.isEnabled.delete` is false, we should force hideCreate/hideDelete to return false
> 
> https://github.com/keystonejs/keystone/blob/3cfc2a3839142dd3ccdbf1dd86768257e9acc0dc/packages/keystone/src/admin-ui/system/getAdminMetaSchema.ts#L204-L227
> 
> . We should also force the itemView fieldMode to read for all the fields of a list when the list has update omitted so you can't get a confusing error by editing and then trying to save (doing that is in the same file as above)
> While we could technically make the creating in the Admin UI work for a list that has querying disabled so arguably the above check should be `isEnabled.type`, we definitely shouldn't do that now so `isEnabled.query` is correct imo.

As discussed will implement this in a separate PR

File: .changeset/twelve-apples-happen.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@keystone-next/keystone': patch
+---
+
+Fixed unqueryable lists being accessible via the Admin UI.
```
Comment (timleslie) [inline .changeset/twelve-apples-happen.md:5]:
```suggestion
Fixed lists with `graphql.omit: ['query']` causing issues in the Admin UI.
```

**Review by mitchellhamilton — APPROVED** [review]

**Pull request merged**